### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by asafgabai**
- [x] All [tests](https://github.com/jfrog/ide-plugins-common/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----
* Class name changes
* Reorganized packages
* Merged GeneralInfo into DependencyNode

### CODE REVIEW

**Review by talarian1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/jfrog/ide/common/components/DependencyNode.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.jfrog.ide.common.utils.Utils.removeComponentIdPrefix;
+
+/**
+ * @author yahavi
+ */
+public class DependencyNode extends ComparableSeverityTreeNode implements Serializable, SubtitledTreeNode {
+    private static final long serialVersionUID = 1L;
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/components/DependencyNode.java:18]: I think we can take the chance and get rid of the Serializable. We use Jackson and not serialization.
```suggestion
public class DependencyNode extends ComparableSeverityTreeNode implements SubtitledTreeNode {
```

File: src/main/java/com/jfrog/ide/common/components/DependencyNode.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    public void setImpactPaths(ImpactTreeNode impactPaths) {
+        this.impactPaths = impactPaths;
+    }
+
+    public void addVulnerabilityOrViolation(IssueNode vulnerabilityOrViolation) {
+        add(vulnerabilityOrViolation);
+    }
+
+    public void sortChildren() {
+        children.sort((treeNode1, treeNode2) -> ((IssueNode) treeNode2).getSeverity().ordinal() - ((IssueNode) treeNode1).getSeverity().ordinal());
```
[the remark this answers is not in the document]
Comment (yahavi) [reply src/main/java/com/jfrog/ide/common/components/DependencyNode.java:72]: I think we can improve this even better - 
Create a common parent that implements the Comparable interface and contains a simplified version of sortChildren method.
Remove all current appearances of sortChildren.

**Review by yahavi — CHANGES_REQUESTED** [review]
Great, @asafgabai! I really like your coding style.

Please consider this:
I noticed that all root classes under "components" are tree nodes. Can we find a better name than the generic "components" name? Like "nodes" or something else. Also if we have subentities, what are the entities?

File: src/main/java/com/jfrog/ide/common/components/DependencyNode.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+import javax.swing.tree.TreeNode;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.jfrog.ide.common.utils.Utils.removeComponentIdPrefix;
+
+/**
+ * @author yahavi
```
[the remark this answers is not in the document]
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/common/components/DependencyNode.java:15]: I removed it.

**Comment by asafgabai** [comment]
> Great, @asafgabai! I really like your coding style.
> 
> Please consider this: I noticed that all root classes under "components" are tree nodes. Can we find a better name than the generic "components" name? Like "nodes" or something else. Also if we have subentities, what are the entities?

I changed it to "nodes". The entities are the nodes.

File: src/main/java/com/jfrog/ide/common/components/DependencyNode.java (hunk 1 above)
Comment (asafgabai) [reply src/main/java/com/jfrog/ide/common/components/DependencyNode.java]: It's going to be used again really soon, when the cache will be back.

File: src/main/java/com/jfrog/ide/common/nodes/SortableChildrenTreeNode.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.jfrog.ide.common.nodes;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import java.util.Comparator;
+
+public class SortableChildrenTreeNode extends DefaultMutableTreeNode {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/nodes/SortableChildrenTreeNode.java:8]: You can move the Comparator implementation declaration to here:
```suggestion
public abstract class SortableChildrenTreeNode<T> extends DefaultMutableTreeNode implements Comparator<T> {
```

By that you would benefit from the following:
1. Remove code duplication - you write `implements Comparator` only once.
2. Enforce implementation of `compareTo` on all child objects.

File: src/main/java/com/jfrog/ide/common/nodes/SortableChildrenTreeNode.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    public void sortChildren() {
+        if (CollectionUtils.isNotEmpty(children)) {
+            children.sort(Comparator.comparing(treeNode -> ((Comparable) treeNode)));
+            children.stream()
+                    .filter(treeNode -> treeNode instanceof SortableChildrenTreeNode)
+                    .forEach(treeNode -> ((SortableChildrenTreeNode) treeNode).sortChildren());
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/common/nodes/SortableChildrenTreeNode.java:14]: Please make sure to handle or suppress all compiler warnings

**Pull request merged**